### PR TITLE
chore: introduce ruff linter and formatter

### PR DESCRIPTION
**Pull request by sambarnes**
gm :v: wanted to check out what u cookin, so did a little housekeeping over coffee this morning

feel free to close if not in line with goals or whatevs

## Details

introduces [ruff](https://docs.astral.sh/ruff) to the codebase -- basically just all the python ecosystem linters in one tool that runs instantly

also added pre-commit hooks that run the suite -- can followup with github actions if theres appetite for it

i dont use vscode, but author recommends [their extension](https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff) for smooth ide integration -- already changed the default formatter in your vscode settings.json

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/OpenRouterTeam/openrouter-runner/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I agree to license this contribution under the MIT LICENSE
- [x] I checked the [current PR](https://github.com/OpenRouterTeam/openrouter-runner/pulls) for duplication.

### CODE REVIEW

**Comment by sambarnes** [comment]
oh and i didnt see any tests so couldnt find a way to quickly sanity check the changes

will have to try taking the whole repo for a spin later

File: modal/runner/engines/vllm.py [hunk 1]
```diff
@@ -89,7 +90,7 @@ async def generate(self, payload: Payload, params):
                 async for request_output in results_generator:
                     final_output = request_output
 
-                output = request_output.outputs[0].text
```
Comment (sambarnes) [inline modal/runner/engines/vllm.py:92]: probs worth reading this specific change a bit closer

only glanced at it, but had a possible scoping bug where request_output was being accessed after the loop terminates

think behavior should be the same with this

**Comment by alexanderatallah** [comment]
@sambarnes this is awesome! have been wanting to try Ruff. @louisgv should review but would love to catch up soon - will ping you

File: modal/runner/engines/vllm.py (hunk 1 above)
Comment (louisgv) [reply modal/runner/engines/vllm.py:92]: Good catch - will run the e2e test without stream on this path, but this diff makes sense to me

**Review by louisgv — APPROVED** [review]
LGTM 👍  - and Merry Christmas 🙌 

Will merge and test it out on dev env

**Pull request merged**